Feature/refactor hydra yaml

ThierryJudge opened the pull request:
**Add support for Hydra**

This PR add supports for Hydra with Yaml configs.

* Adds hydra config for segmentation and classification systems for mlp, unet and enet. 
* Adds support for python-dotenv to load system specific environment variables
* Adds examples on how to use vital submodule and hydra configs in other projects. 

Still needs to be done (in subsequent PRs): 
* Refactor readme
* Allow optimizer configuration
* Add missing configs for networks
* Add support for auto-encoders

nathanpainchaud committed:
Refactored DataManagerMixin as LightningDataModule

ThierryJudge committed:
working on hydra to replace argparse

ThierryJudge committed:
working training for mnist with "python runner.py +data=mnist +system=classification +module=mlp"

ThierryJudge committed:
Merge branch 'dev' of https://github.com/nathanpainchaud/vital into tmp/refactor/hydra

ThierryJudge committed:
working training for acdc

ThierryJudge committed:
Add callbacks

ThierryJudge committed:
change callbacks from list to dict

ThierryJudge committed:
add check_cfg method

ThierryJudge committed:
add check for default root dir

ThierryJudge committed:
implement yaml configs

ThierryJudge committed:
change config dir name,
log hparams

ThierryJudge committed:
callbacks and config

ThierryJudge committed:
fix callbacks and loggers
add arguments for system instantiate
add missing params to camus config

ThierryJudge committed:
Fix pre-commit

ThierryJudge committed:
merge dev

ThierryJudge committed:
add hydra to requirements.txt
fix camus labels

ThierryJudge committed:
allow more configurable callbacks

ThierryJudge committed:
fix loading from checkpoint path, add hydra and dotenv to requirements.txt

ThierryJudge committed:
Merge branch 'tmp/hydra/yaml' of https://github.com/nathanpainchaud/vital into tmp/hydra/yaml

ThierryJudge committed:
allow model saving if not checkpoint if used

ThierryJudge committed:
fix loggers

ThierryJudge committed:
Move module configs in system folder to allow system config to override module parameters

ThierryJudge committed:
Work in progress:
* Add softmax and mcdropout system
* Add evaluation pipeline (Seg metrics, calibration overlap)

ThierryJudge committed:
change unet dropout parameter name to be uniform with enet

ThierryJudge committed:
Add resolvers for num_workers and gpus

ThierryJudge committed:
add workers=True to seed_everything in runner.py

ThierryJudge committed:
Merge dev.

ThierryJudge committed:
Remove argparse methods.

lemairecarl reviewed (commented):
Looks good, great work! Not yet finished, but I like the direction this is going into.

BTW, why does this other PR (still) exist? https://github.com/nathanpainchaud/vital/pull/19

ThierryJudge committed:
Minor fixes and PR comment fixes

ThierryJudge committed:
Remove test_step from computation.
Remove resume/train/ckpt_path errors from runner
Change default name to remove error.

ThierryJudge committed:
minor fixes

ThierryJudge committed:
Merge branch 'feature/refactor-hydra-yaml' of https://github.com/nathanpainchaud/vital into feature/refactor-hydra-yaml

ThierryJudge committed:
Remove log images
Update gitignore
add monitor to model_checkpoint.yaml to avoid warning

nathanpainchaud committed:
Loading checkpoint from Comet model registry

ThierryJudge committed:
Merge dev

nathanpainchaud committed:
Refactored checkpoint CLI to document Comet model registry option

nathanpainchaud committed:
Fixed minimum Comet version to make both version and stage optional

nathanpainchaud committed:
Implemented caching logic for Comet checkpoints

nathanpainchaud committed:
Changed comet integration to rely on env vars instead of config file

nathanpainchaud committed:
Refactored Comet API init to support any API key setting method recognised by Comet

ThierryJudge committed:
Merge feature/comet-model-registry into hydra refactor

ThierryJudge committed:
Add comet tags and logging .hydra dir to comet.
Add comment regarding use of local ckpt with relative path

ThierryJudge committed:
Update .gitignore

nathanpainchaud reviewed (changes requested):
Some minor final comments that can be fixed very easily. Otherwise, just 2 general things to do before merging:

1. I pushed some fixes for the Comet model registry query to the main dev branch. I suggest you merge those changes first;
2. Seems you're in conflict with the hydra branch you're merging into, so you should fix those!

ThierryJudge committed:
Merge dev into feature/refactor-hydra-yaml

ThierryJudge committed:
PR comment fixes.

ThierryJudge committed:
Merge branch 'feature/refactor-hydra-yaml' of https://github.com/nathanpainchaud/vital into feature/refactor-hydra-yaml

ThierryJudge committed:
Minor fix

nathanpainchaud committed:
Refactored MLP to make it more configurable

nathanpainchaud committed:
Broaden accepted values for label to use for CAMUS

- Also added new method `DataTag` to simplify listing all possible enum elements

ThierryJudge committed:
Add hydra-dev to CI workflows.

lemairecarl reviewed (commented):
Oops, I "missed the deadline" of the merge of this PR! Sorry. (I was reviewing when it was merged.)

Anyways, it looks all good to me, except for one small thing.

Great work guys!
